src/iinit.cpp

```diff
@@ -66,11 +69,225 @@ mkrodsdir() {
+        // environment has a value already, the default value is the existing value (meaning, there will be no
+        // change in the environment). Therefore, the else case is a no-op as no change will occur.
+        if (!response.empty()) {
+            try {
+                _setting = boost::lexical_cast<int>(response);
+            }
+            catch (const boost::bad_lexical_cast&) {
+                _setting = default_value;
+            }
+        }
+        else if (!env_has_value) {
+            _setting = default_value;
+        }
+    } // set_env_from_prompt
+
+    auto configure_required_settings_in_env(RodsEnvironment& _env, nlohmann::json& _json_env) -> void
+    {
+        if (0 == std::strlen(_env.rodsHost)) {
+            constexpr const char* host_prompt = "Enter the host name (DNS) of the server to connect to";
+            set_env_from_prompt(_env.rodsHost, host_prompt, sizeof(_env.rodsHost));
+            _json_env[irods::KW_CFG_IRODS_HOST] = _env.rodsHost;
+        }
+
+        if (0 == _env.rodsPort) {
+            constexpr const char* port_prompt = "Enter the port number";
+            constexpr int default_port = 1247;
+            set_env_from_prompt(_env.rodsPort, default_port, port_prompt);
+            _json_env[irods::KW_CFG_IRODS_PORT] = _env.rodsPort;
+        }
+
+        if (0 == std::strlen(_env.rodsUserName)) {
+            constexpr const char* username_prompt = "Enter your iRODS user name";
+            set_env_from_prompt(_env.rodsUserName, username_prompt, sizeof(_env.rodsUserName));
+            _json_env[irods::KW_CFG_IRODS_USER_NAME] = _env.rodsUserName;
+        }
+
+        if (0 == std::strlen(_env.rodsZone)) {
+            constexpr const char* zone_prompt = "Enter your iRODS zone";
+            set_env_from_prompt(_env.rodsZone, zone_prompt, sizeof(_env.rodsZone));
+            _json_env[irods::KW_CFG_IRODS_ZONE] = _env.rodsZone;
+        }
+    } // configure_required_settings_in_env
+
+    auto configure_ssl_in_env(RodsEnvironment& _env, nlohmann::json& _json_env) -> void
+    {
+        // If the user indicated that SSL is going to be used, these settings are required, so no prompts occur.
+        constexpr const char* default_client_server_policy = "CS_NEG_REQUIRE";
+        std::strncpy(_env.rodsClientServerPolicy, default_client_server_policy, sizeof(_env.rodsClientServerPolicy));
+        _json_env[irods::KW_CFG_IRODS_CLIENT_SERVER_POLICY] = _env.rodsClientServerPolicy;
+
+        constexpr const char* default_server_negotiation = "request_server_negotiation";
+        std::strncpy(
+            _env.rodsClientServerNegotiation, default_server_negotiation, sizeof(_env.rodsClientServerNegotiation));
+        _json_env[irods::KW_CFG_IRODS_CLIENT_SERVER_NEGOTIATION] = _env.rodsClientServerNegotiation;
+
+        constexpr const char* default_server_verification = "cert";
+        constexpr const char* server_verification_prompt = "Enter the server verification level";
```

Comment:
consider showing a bracketed set of choices... the enum...
"Enter the server verification level [none, cert, hostname]"

also, consider that we say in the docs that 'hostname' is the default if it's not set... so maybe we should default to hostname here too?

Reply:
Showing the choices makes me wonder why we wouldn't show choices (where applicable) for other questions (auth scheme, encryption algorithm, etc.). Plus, the behavior here is to use what's in the brackets as the default value if nothing is entered. How should we indicate the default value if we show the options (if indeed there is still a default)?

Didn't notice that "hostname" is the default. Will change to that once we decide on the above

Reply:
There are two ways we can go about this:
1. Show all choices available to the user in each prompt. I would think this should be considered a separate unit of work. We do not have strings for that stuff readily available in a library function for all of these prompts. Plus, we would need to consider what's available to the client versus the server, etc.
2. Show only the default choice to users when there is a default for a particular prompt, in which case nothing is shown (how it behaves now)

I think either option is fine, but I think if we go with option 1, we should make a new issue and move along.

As before, I'll fix this case so that it says "default".

Reply:
The default has been changed to "hostname".

I created https://github.com/irods/irods/issues/7142 to continue this discussion. For now, I'm going to just show the default values and we can change it before release. If we would rather show nothing, just open a new thread.